src/App.tsx

```diff
@@ -1,24 +1,147 @@
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
+import { createTodo, deleteTodo, getTodos } from './api/todos';
+import { Header } from './Components/Header';
+import { Main } from './Components/Main';
+import { Footer } from './Components/Footer';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { Errors } from './types/Errors';
 
-const USER_ID = 0;
+const USER_ID = 10307;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  // if (!USER_ID) {
+  //   return <UserWarning />;
+  // }
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errors, setErrors] = useState<Errors | null>(null);
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [waitingRespone, setWaitingResponse] = useState<number[]>([]);
+
+  const addTodo = (todoTitle: string) => {
+    const todo: Todo = {
+      title: todoTitle,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    setTempTodo({
+      id: 0,
+      title: todoTitle,
+      userId: USER_ID,
+      completed: false,
+    });
+
+    createTodo(todo)
+      .then((value) => {
+        setTodos([...todos, value]);
+        setTempTodo(null);
+      })
+      .catch(() => {
+        setErrors({ posting: true });
+        setTempTodo(null);
+      });
+  };
+
+  const removeTodo = (id: number) => {
+    setWaitingResponse([id]);
+    deleteTodo(id)
+      .then(() => {
+        setTodos(todos.filter((todo) => todo.id !== id));
+        setWaitingResponse([]);
+      })
+      .catch(() => {
+        setErrors({ deleting: true });
+      });
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((todosList) => {
+        setTodos(todosList);
+      })
+      .catch(() => {
+        setErrors({ loading: true });
+      });
+  }, []);
+
+  const filterTodos = todos.filter((todo) => {
+    switch (filter) {
+      case Filter.All:
+        return todo;
+
+      case Filter.Active:
+        return !todo.completed;
+
+      case Filter.Completed:
+        return todo.completed;
+
+      default:
+        return todo;
+    }
+  });
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          addTodo={addTodo}
+          todos={todos}
+          setTodos={setTodos}
+          setErrors={setErrors}
+          tempTodo={tempTodo}
+          setWaitingResponse={setWaitingResponse}
```

Comment:
Try to do it in all cases
```suggestion
          onAdd={addTodo}
          todos={todos}
          onChange={setTodos}
          onError={setErrors}
          tempTodo={tempTodo}
          onLoading={setWaitingResponse}
```